source/System.cpp

```diff
@@ -513,6 +513,13 @@ void System::UpdateSystem(const Set<System> &systems, const set<double> &neighbo
 		attributes.erase("uninhabited");
 	else
 		attributes.insert("uninhabited");
+
+	// Calculate the smallest arrival period of a fleet (or 0 if no fleets arrive)
+	minimumFleetPeriod = numeric_limits<int>::max();
+	for(auto &event : fleets)
+		minimumFleetPeriod = min<int>(minimumFleetPeriod, event.Period());
+	if(minimumFleetPeriod == numeric_limits<int>::max())
+		minimumFleetPeriod = 0;
```

Comment:
Can this not be done at the end of `System::Load()`?

Reply:
From this comment in the code, it seems UpdateSystem is where it belongs:

```
// Update any information about the system that may have changed due to events,
// or because the game was started, e.g. neighbors, solar wind and power, or
// if the system is inhabited.
void System::UpdateSystem(const Set<System> &systems, const set<double> &neighborDistances)
```